GoogleForCreators/web-stories-wp
# Dashboard: Telemetry Opt In Banner

## Summary

This adds an opt in banner for telemetry tracking that will display until the user either closes it or interacts with the telemetry opt in checkbox on the `Edit Settings` page/view. It will be visible on all top level views using the `PageHeader` shared component with the exception of the `Edit Settings` page/view.
## Relevant Technical Choices

- I wrote a `useTelemetryOptIn` hook so as not to duplicate logic between the banner and settings page.
- I created a container component for the banner in the same file so we would not have to add bunch of `telemetry` props to the `PageHeader` component and end up editing all of our top level views.
- I added the `@testing-library/user-events` dependency to circumvent the following issue: https://github.com/testing-library/react-testing-library/issues/275
- I added a `MockApiProvider` and `renderWithProvider` utils to make view integration tests easier.

## User-facing changes

- This adds a banner to the top of the page of both My Stories and Explore Templates that allows the user to opt in or close.  Once closed or opted in on the `Editor Settings page`, the banner should not display again.  In addition, when opting in using the banner the title should change to `Your selection has been updated. Thank you for helping to improve the editor!`.


![banner](https://user-images.githubusercontent.com/2755722/92417108-90e5f280-f12e-11ea-93e3-e8cbb0e2e7a6.png)


## Testing Instructions

- Verify that the unit and karma tests pass
- Verify that the banner does not reappear after closing it
- Verify that opting in using the banner changes the title header to `Your selection has been updated. Thank you for helping to improve the editor.` and unchecking it reverts it back.

#4226
